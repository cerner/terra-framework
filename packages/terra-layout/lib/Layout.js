"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _bind = _interopRequireDefault(require("classnames/bind"));
var _terraContentContainer = _interopRequireDefault(require("terra-content-container"));
var _lodash = _interopRequireDefault(require("lodash.debounce"));
var _LayoutSlidePanelModule = _interopRequireDefault(require("./LayoutSlidePanel.module.scss"));
var _LayoutSlidePanel = _interopRequireDefault(require("./_LayoutSlidePanel"));
var _LayoutUtils = require("./LayoutUtils");
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
var cx = _bind.default.bind(_LayoutSlidePanelModule.default);
var propTypes = {
  /**
   * Element to be placed within the header section of the layout.
   */
  header: _propTypes.default.element,
  /**
   * Element to be placed within the menu section of the layout. If not set, Layout-provided menu controls will be hidden.
   */
  menu: _propTypes.default.element,
  /**
   * String used to decorate menu hover control.
   */
  menuText: _propTypes.default.string,
  /**
   * Element to be placed within the main content section of the layout.
   */
  children: _propTypes.default.element
};
var Layout = /*#__PURE__*/function (_React$Component) {
  (0, _inherits2.default)(Layout, _React$Component);
  var _super = _createSuper(Layout);
  function Layout(props) {
    var _this;
    (0, _classCallCheck2.default)(this, Layout);
    _this = _super.call(this, props);
    _this.toggleMenu = _this.toggleMenu.bind((0, _assertThisInitialized2.default)(_this));
    _this.togglePin = _this.togglePin.bind((0, _assertThisInitialized2.default)(_this));
    _this.updateSize = (0, _lodash.default)(_this.updateSize.bind((0, _assertThisInitialized2.default)(_this)), 100);
    _this.renderHeader = _this.renderHeader.bind((0, _assertThisInitialized2.default)(_this));
    _this.renderMenu = _this.renderMenu.bind((0, _assertThisInitialized2.default)(_this));
    _this.renderContent = _this.renderContent.bind((0, _assertThisInitialized2.default)(_this));
    _this.state = Layout.stateForProps(props, {
      size: (0, _LayoutUtils.getBreakpointSize)(),
      prevProps: _this.props
    });
    return _this;
  }
  (0, _createClass2.default)(Layout, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      window.addEventListener('resize', this.updateSize);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      window.removeEventListener('resize', this.updateSize);
    }
  }, {
    key: "updateSize",
    value: function updateSize() {
      var newSize = (0, _LayoutUtils.getBreakpointSize)();
      if (this.state.size !== newSize) {
        this.setState(Layout.stateForProps(this.props, {
          size: newSize
        }));
      }
    }
  }, {
    key: "toggleMenu",
    value: function toggleMenu() {
      var _this2 = this;
      return new Promise(function (resolve) {
        _this2.setState(function (prevState) {
          return {
            menuIsOpen: !prevState.menuIsOpen
          };
        }, resolve);
      });
    }
  }, {
    key: "togglePin",
    value: function togglePin() {
      var _this3 = this;
      return new Promise(function (resolve) {
        _this3.setState(function (prevState) {
          return {
            menuIsPinned: !prevState.menuIsPinned
          };
        }, resolve);
      });
    }
  }, {
    key: "renderHeader",
    value: function renderHeader() {
      var header = this.props.header;
      var _this$state = this.state,
        size = _this$state.size,
        menuIsOpen = _this$state.menuIsOpen,
        isToggleMenu = _this$state.isToggleMenu,
        menuIsPresent = _this$state.menuIsPresent;
      if (!header) {
        return null;
      }
      var shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
      return /*#__PURE__*/_react.default.cloneElement(header, {
        layoutConfig: {
          size: size,
          toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
          menuIsOpen: menuIsOpen
        }
      });
    }
  }, {
    key: "renderMenu",
    value: function renderMenu() {
      var menu = this.props.menu;
      var _this$state2 = this.state,
        size = _this$state2.size,
        menuIsOpen = _this$state2.menuIsOpen,
        menuIsPinned = _this$state2.menuIsPinned,
        isToggleMenu = _this$state2.isToggleMenu,
        menuIsPresent = _this$state2.menuIsPresent;
      var shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
      if (!menuIsPresent) {
        return null;
      }
      return /*#__PURE__*/_react.default.cloneElement(menu, {
        layoutConfig: {
          size: size,
          toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
          menuIsOpen: menuIsOpen,
          menuIsPinned: menuIsPinned
        }
      });
    }
  }, {
    key: "renderContent",
    value: function renderContent() {
      var children = this.props.children;
      var _this$state3 = this.state,
        size = _this$state3.size,
        menuIsOpen = _this$state3.menuIsOpen,
        isToggleMenu = _this$state3.isToggleMenu,
        menuIsPresent = _this$state3.menuIsPresent;
      var shouldAllowMenuToggle = isToggleMenu && menuIsPresent;
      return /*#__PURE__*/_react.default.createElement(_terraContentContainer.default, {
        fill: true,
        header: isToggleMenu && this.renderHeader(),
        className: cx('content-container')
      }, children ? /*#__PURE__*/_react.default.cloneElement(children, {
        layoutConfig: {
          size: size,
          toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
          menuIsOpen: menuIsOpen
        }
      }) : null);
    }
  }, {
    key: "render",
    value: function render() {
      var menuText = this.props.menuText;
      var _this$state4 = this.state,
        menuIsOpen = _this$state4.menuIsOpen,
        menuIsPinned = _this$state4.menuIsPinned,
        size = _this$state4.size,
        isFixedMenu = _this$state4.isFixedMenu,
        isToggleMenu = _this$state4.isToggleMenu;
      return /*#__PURE__*/_react.default.createElement(_terraContentContainer.default, (0, _extends2.default)({
        fill: true,
        header: !isToggleMenu && this.renderHeader()
      }, (0, _LayoutUtils.getCustomProps)(this.props, propTypes)), /*#__PURE__*/_react.default.createElement(_LayoutSlidePanel.default, {
        panelContent: this.renderMenu(),
        panelBehavior: menuIsPinned || isFixedMenu ? 'squish' : 'overlay',
        size: size,
        onToggle: this.toggleMenu,
        toggleText: menuText,
        isOpen: menuIsOpen,
        isAnimated: true
      }, this.renderContent()));
    }
  }], [{
    key: "stateForProps",
    value: function stateForProps(props, currentState) {
      var isToggleMenu = currentState.size === 'tiny' || currentState.size === 'small';
      var isFixedMenu = !isToggleMenu;
      var menuIsPresent = !!props.menu;
      return _objectSpread(_objectSpread({}, currentState || {}), {}, {
        isFixedMenu: isFixedMenu,
        isToggleMenu: isToggleMenu,
        menuIsPresent: menuIsPresent,
        menuIsOpen: menuIsPresent && (currentState.menuIsOpen || isFixedMenu),
        menuIsPinned: menuIsPresent && currentState.menuIsPinned
      });
    }
  }, {
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(nextProps, prevState) {
      if (nextProps !== prevState.prevProps) {
        return Layout.stateForProps(nextProps, prevState);
      }
      return null;
    }
  }]);
  return Layout;
}(_react.default.Component);
Layout.propTypes = propTypes;
var _default = Layout;
exports.default = _default;