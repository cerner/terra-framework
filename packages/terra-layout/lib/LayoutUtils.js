"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getCustomProps = exports.getBreakpointSize = void 0;
var breakpoints = {
  tiny: 544,
  small: 768,
  medium: 992,
  large: 1216,
  huge: 1440
};
var small = breakpoints.small,
  medium = breakpoints.medium,
  large = breakpoints.large,
  huge = breakpoints.huge;
var getBreakpointSize = function getBreakpointSize(queryWidth) {
  var width = queryWidth || window.innerWidth;
  if (width >= huge) {
    return 'huge';
  }
  if (width >= large) {
    return 'large';
  }
  if (width >= medium) {
    return 'medium';
  }
  if (width >= small) {
    return 'small';
  }
  return 'tiny';
};
exports.getBreakpointSize = getBreakpointSize;
var getCustomProps = function getCustomProps(props, validPropTypes) {
  return Object.keys(props).filter(function (key) {
    return !Object.keys(validPropTypes).includes(key);
  }).reduce(function (customProps, key) {
    customProps[key] = props[key]; // eslint-disable-line no-param-reassign
    return customProps;
  }, {});
};
exports.getCustomProps = getCustomProps;