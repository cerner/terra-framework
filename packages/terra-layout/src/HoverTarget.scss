:local {
  .hover-target {
    overflow: hidden;
    width: 100%;
  }

  .content-section {
    float: left;
    -webkit-overflow-scrolling: touch;
    -ms-overflow-style: -ms-autohiding-scrollbar;
    overflow-y: auto;
    width: calc(100% - var(--layout-menu-hover-target-width, 2.143rem));
  }

  .content-section[aria-hidden='true'] {
    visibility: hidden;
  }

  .hover-section {
    background: var(--layout-menu-hover-target-background-color, #e8e9ea);
    border: 0;
    border-right: var(--layout-menu-hover-target-border-right, 0.07143rem solid #bcbfc0);
    float: right;
    overflow: hidden;
    position: relative;
    width: var(--layout-menu-hover-target-width, 2.143rem);

    &:active {
      background: var(--layout-menu-hover-target-background-color, #c6c7c9);
    }
  }

  .hover-section-wrapper {
    align-items: stretch;
    display: flex;
    flex-direction: column;
    height: 100%;
    overflow: hidden;
  }

  .hover-section-icon {
    flex: 0 0 auto;
    padding-top: 0.7143rem;
    width: 100%;
  }

  .hover-section-text {
    flex: 1 1 auto;
    width: 100%;
  }

  .hover-text {
    color: var(--layout-menu-hover-target-text-color, #4e5558);
    float: left;
    font-size: var(--layout-menu-hover-target-font-size, 0.857rem);
    left: 50%;
    margin: 0;
    overflow: hidden;
    padding: 0 0.35714rem 0 1.0714rem;
    position: relative;
    text-align: right;
    text-overflow: ellipsis;
    transform: rotate(-90deg) translate3d(-100%, -50%, 0);
    transform-origin: left top 0;
    vertical-align: middle;
    white-space: nowrap;
    word-break: normal;
    word-wrap: normal;
  }

  .hover-icon {
    color: var(--layout-menu-hover-target-icon-color, #909496);
    font-size: 0.571rem;
    position: relative;
    width: 100%;
  }

  .hover-target,
  .content-section,
  .hover-section {
    height: 100%;
    position: relative;
  }

  .is-hover-disabled,
  .is-hover-tab-hidden {
    > .content-section {
      width: 100%;
    }

    > .hover-section {
      display: none;
      overflow: hidden;
      width: 0;
    }
  }
}
