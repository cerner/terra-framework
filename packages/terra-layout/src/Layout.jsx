import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import ContentContainer from 'terra-content-container';
import LodashDebounce from 'lodash.debounce';

import styles from './LayoutSlidePanel.module.scss';
import LayoutSlidePanel from './_LayoutSlidePanel';
import {
  getBreakpointSize,
  getCustomProps,
} from './LayoutUtils';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Element to be placed within the header section of the layout.
   */
  header: PropTypes.element,
  /**
   * Element to be placed within the menu section of the layout. If not set, Layout-provided menu controls will be hidden.
   */
  menu: PropTypes.element,
  /**
   * String used to decorate menu hover control.
   */
  menuText: PropTypes.string,
  /**
   * Element to be placed within the main content section of the layout.
   */
  children: PropTypes.element,
};

class Layout extends React.Component {
  static stateForProps(props, currentState) {
    const isToggleMenu = currentState.size === 'tiny' || currentState.size === 'small';
    const isFixedMenu = !isToggleMenu;

    const menuIsPresent = !!props.menu;

    return {
      ...currentState || {},
      isFixedMenu,
      isToggleMenu,
      menuIsPresent,
      menuIsOpen: menuIsPresent && (currentState.menuIsOpen || isFixedMenu),
      menuIsPinned: menuIsPresent && currentState.menuIsPinned,
    };
  }

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps !== prevState.prevProps) {
      return Layout.stateForProps(nextProps, prevState);
    }

    return null;
  }

  constructor(props) {
    super(props);

    this.toggleMenu = this.toggleMenu.bind(this);
    this.togglePin = this.togglePin.bind(this);
    this.updateSize = LodashDebounce(this.updateSize.bind(this), 100);
    this.renderHeader = this.renderHeader.bind(this);
    this.renderMenu = this.renderMenu.bind(this);
    this.renderContent = this.renderContent.bind(this);

    this.state = Layout.stateForProps(props, {
      size: getBreakpointSize(),
      prevProps: this.props,
    });
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateSize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateSize);
  }

  updateSize() {
    const newSize = getBreakpointSize();

    if (this.state.size !== newSize) {
      this.setState(Layout.stateForProps(this.props, {
        size: newSize,
      }));
    }
  }

  toggleMenu() {
    return new Promise((resolve) => {
      this.setState(prevState => ({
        menuIsOpen: !prevState.menuIsOpen,
      }), resolve);
    });
  }

  togglePin() {
    return new Promise((resolve) => {
      this.setState(prevState => ({
        menuIsPinned: !prevState.menuIsPinned,
      }), resolve);
    });
  }

  renderHeader() {
    const { header } = this.props;
    const {
      size, menuIsOpen, isToggleMenu, menuIsPresent,
    } = this.state;

    if (!header) {
      return null;
    }

    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;

    return React.cloneElement(header, {
      layoutConfig: {
        size,
        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
        menuIsOpen,
      },
    });
  }

  renderMenu() {
    const { menu } = this.props;
    const {
      size, menuIsOpen, menuIsPinned, isToggleMenu, menuIsPresent,
    } = this.state;
    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;

    if (!menuIsPresent) {
      return null;
    }

    return React.cloneElement(menu, {
      layoutConfig: {
        size,
        toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
        menuIsOpen,
        menuIsPinned,
      },
    });
  }

  renderContent() {
    const { children } = this.props;
    const {
      size, menuIsOpen, isToggleMenu, menuIsPresent,
    } = this.state;
    const shouldAllowMenuToggle = isToggleMenu && menuIsPresent;

    return (
      <ContentContainer
        fill
        header={isToggleMenu && this.renderHeader()}
        className={cx('content-container')}
      >
        {children
          ? React.cloneElement(children, {
            layoutConfig: {
              size,
              toggleMenu: shouldAllowMenuToggle ? this.toggleMenu : undefined,
              menuIsOpen,
            },
          }) : null}
      </ContentContainer>
    );
  }

  render() {
    const { menuText } = this.props;
    const {
      menuIsOpen, menuIsPinned, size, isFixedMenu, isToggleMenu,
    } = this.state;

    return (
      <ContentContainer
        fill
        header={!isToggleMenu && this.renderHeader()}
        {...getCustomProps(this.props, propTypes)}
      >
        <LayoutSlidePanel
          panelContent={this.renderMenu()}
          panelBehavior={menuIsPinned || isFixedMenu ? 'squish' : 'overlay'}
          size={size}
          onToggle={this.toggleMenu}
          toggleText={menuText}
          isOpen={menuIsOpen}
          isAnimated
        >
          {this.renderContent()}
        </LayoutSlidePanel>
      </ContentContainer>
    );
  }
}

Layout.propTypes = propTypes;

export default Layout;
