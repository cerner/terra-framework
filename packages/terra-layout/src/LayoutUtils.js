const breakpoints = {
  tiny: 544,
  small: 768,
  medium: 992,
  large: 1216,
  huge: 1440,
};

const {
  small, medium, large, huge,
} = breakpoints;

const getBreakpointSize = (queryWidth) => {
  const width = queryWidth || window.innerWidth;
  if (width >= huge) {
    return 'huge';
  } if (width >= large) {
    return 'large';
  } if (width >= medium) {
    return 'medium';
  } if (width >= small) {
    return 'small';
  }
  return 'tiny';
};

const getCustomProps = (props, validPropTypes) => (
  Object.keys(props)
    .filter(key => !Object.keys(validPropTypes).includes(key))
    .reduce((customProps, key) => {
      customProps[key] = props[key]; // eslint-disable-line no-param-reassign
      return customProps;
    }, {})
);

export {
  getBreakpointSize,
  getCustomProps,
};
