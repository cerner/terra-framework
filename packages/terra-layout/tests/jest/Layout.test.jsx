import React from 'react';

import Layout from '../../src/Layout';
import { getBreakpointSize, getCustomProps } from '../../src/LayoutUtils';

jest.mock('../../src/LayoutUtils');

describe('Layout', () => {
  it('should render a Layout without optional props', () => {
    getBreakpointSize.mockImplementation(() => 'tiny');

    const result = enzyme.shallow((
      <Layout />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render a Layout with custom props', () => {
    getBreakpointSize.mockImplementation(() => 'tiny');
    getCustomProps.mockReturnValueOnce({ 'data-test-attr': 'test-val' });

    const result = enzyme.shallow((
      <Layout />
    ));
    expect(result).toMatchSnapshot();
  });

  describe('Menu Disabled', () => {
    const layout = (
      <Layout
        header={<div>Test Header</div>}
      >
        <div>Test Content</div>
      </Layout>
    );

    it('should render a Layout when tiny', () => {
      getBreakpointSize.mockImplementation(() => 'tiny');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when small', () => {
      getBreakpointSize.mockImplementation(() => 'small');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when medium', () => {
      getBreakpointSize.mockImplementation(() => 'medium');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when large', () => {
      getBreakpointSize.mockImplementation(() => 'large');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when huge', () => {
      getBreakpointSize.mockImplementation(() => 'huge');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });
  });

  describe('Menu Enabled', () => {
    const layout = (
      <Layout
        header={<div>Test Header</div>}
        menu={<div>Test Menu</div>}
        menuText="Menu Text"
      >
        <div>Test Content</div>
      </Layout>
    );

    it('should render a Layout when tiny', () => {
      getBreakpointSize.mockImplementation(() => 'tiny');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when small', () => {
      getBreakpointSize.mockImplementation(() => 'small');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when medium', () => {
      getBreakpointSize.mockImplementation(() => 'medium');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when large', () => {
      getBreakpointSize.mockImplementation(() => 'large');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });

    it('should render a Layout when huge', () => {
      getBreakpointSize.mockImplementation(() => 'huge');

      const result = enzyme.shallow(layout);
      expect(result).toMatchSnapshot();
    });
  });
});
