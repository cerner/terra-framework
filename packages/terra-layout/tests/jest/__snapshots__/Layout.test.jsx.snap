// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Menu Disabled should render a Layout when huge 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": false,
          "size": "huge",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={null}
    size="huge"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "huge",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when large 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": false,
          "size": "large",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={null}
    size="large"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "large",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when medium 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": false,
          "size": "medium",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={null}
    size="medium"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "medium",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when small 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={null}
    size="small"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={
        <div
          layoutConfig={
            Object {
              "menuIsOpen": false,
              "size": "small",
              "toggleMenu": undefined,
            }
          }
        >
          Test Header
        </div>
      }
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "small",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Disabled should render a Layout when tiny 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={null}
    size="tiny"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={
        <div
          layoutConfig={
            Object {
              "menuIsOpen": false,
              "size": "tiny",
              "toggleMenu": undefined,
            }
          }
        >
          Test Header
        </div>
      }
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "tiny",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when huge 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": true,
          "size": "huge",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={true}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "menuIsPinned": undefined,
            "size": "huge",
            "toggleMenu": undefined,
          }
        }
      >
        Test Menu
      </div>
    }
    size="huge"
    toggleText="Menu Text"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "size": "huge",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when large 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": true,
          "size": "large",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={true}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "menuIsPinned": undefined,
            "size": "large",
            "toggleMenu": undefined,
          }
        }
      >
        Test Menu
      </div>
    }
    size="large"
    toggleText="Menu Text"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "size": "large",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when medium 1`] = `
<ContentContainer
  fill={true}
  header={
    <div
      layoutConfig={
        Object {
          "menuIsOpen": true,
          "size": "medium",
          "toggleMenu": undefined,
        }
      }
    >
      Test Header
    </div>
  }
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={true}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="squish"
    panelContent={
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "menuIsPinned": undefined,
            "size": "medium",
            "toggleMenu": undefined,
          }
        }
      >
        Test Menu
      </div>
    }
    size="medium"
    toggleText="Menu Text"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={false}
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": true,
            "size": "medium",
            "toggleMenu": undefined,
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when small 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "menuIsPinned": undefined,
            "size": "small",
            "toggleMenu": [Function],
          }
        }
      >
        Test Menu
      </div>
    }
    size="small"
    toggleText="Menu Text"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={
        <div
          layoutConfig={
            Object {
              "menuIsOpen": false,
              "size": "small",
              "toggleMenu": [Function],
            }
          }
        >
          Test Header
        </div>
      }
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "small",
            "toggleMenu": [Function],
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout Menu Enabled should render a Layout when tiny 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "menuIsPinned": undefined,
            "size": "tiny",
            "toggleMenu": [Function],
          }
        }
      >
        Test Menu
      </div>
    }
    size="tiny"
    toggleText="Menu Text"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={
        <div
          layoutConfig={
            Object {
              "menuIsOpen": false,
              "size": "tiny",
              "toggleMenu": [Function],
            }
          }
        >
          Test Header
        </div>
      }
    >
      <div
        layoutConfig={
          Object {
            "menuIsOpen": false,
            "size": "tiny",
            "toggleMenu": [Function],
          }
        }
      >
        Test Content
      </div>
    </ContentContainer>
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout should render a Layout with custom props 1`] = `
<ContentContainer
  data-test-attr="test-val"
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={null}
    size="tiny"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={null}
    />
  </LayoutSlidePanel>
</ContentContainer>
`;

exports[`Layout should render a Layout without optional props 1`] = `
<ContentContainer
  fill={true}
  header={false}
>
  <LayoutSlidePanel
    isAnimated={true}
    isOpen={false}
    isToggleEnabled={false}
    onToggle={[Function]}
    panelBehavior="overlay"
    panelContent={null}
    size="tiny"
  >
    <ContentContainer
      className="content-container"
      fill={true}
      header={null}
    />
  </LayoutSlidePanel>
</ContentContainer>
`;
